client/src/components/base-editor/base-editor.tsx

```diff
@@ -24,24 +27,94 @@ const BaseEditor = forwardRef(
     ref: React.Ref<MonacoEditor>,
   ) => {
     const { textEditorSettings } = useContext(SettingsContext)
+    const { customBinding, theme } = textEditorSettings
+
+    const boundEditor = useRef<any>(undefined)
+    const editor = useRef<monaco.editor.IStandaloneCodeEditor | undefined>(
+      undefined,
+    )
+    const actionsContainer = useRef<HTMLDivElement | null>(null)
+
+    useEffect(() => {
+      if (customBinding === 'none' && boundEditor.current) {
+        boundEditor.current.dispose()
+      } else if (
+        customBinding === 'vim' &&
+        editor.current &&
+        actionsContainer.current
+      ) {
+        // if switching directly from emacs
+        if (boundEditor.current) {
+          boundEditor.current.dispose()
+        }
+
+        boundEditor.current = initVimMode(
+          editor.current,
+          actionsContainer.current,
+        )
+      } else if (
+        customBinding === 'emacs' &&
+        editor.current &&
+        actionsContainer.current
+      ) {
+        // if switching directly from vim
+        boundEditor.current && boundEditor.current.dispose()
+
+        boundEditor.current = new EmacsExtension(editor.current)
```

Comment:
Let's abstract the `EmacsExtension` initialization into a function `initEmacsMode`. This way we could have all code  with the same level of abstraction.